xtas/tasks/pipeline.py

```diff
@@ -22,34 +22,42 @@ def pipeline(doc, pipeline, store_final=True, store_intermediate=False,
     @param store_final: if True, store the final result
     @param store_intermediate: if True, store all intermediate results as well
     """
-    # form basic pipeline by resolving task dictionaries to task objects
-    tasks = [_get_task(t) for t in pipeline]
 
+    tasks = map(_get_task, pipeline)
```

Comment:
List comprehensions are more "Pythonic". Also, map's behavior changes in Python 3: it returns an iterable rather than a list, so you get things like

```py
>>> it = map(str, [1, 2, 3])
>>> it
<map object at 0x7f8984df7208>
>>> len(it)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: object of type 'map' has no len()
```

We don't support Py3 because not all dependencies do, but I still want the code to be future-proof. I just reverted 4b8b4cd55c8c6670acd75b07b2f4da1a01a6b781 for the same reason.